plugins/Participated/class.participated.plugin.php

```diff
@@ -1,202 +1,247 @@
+            ->join('UserDiscussion w', "d.DiscussionID = w.DiscussionID and w.UserID = {$userID}", 'left')
+            ->join('Comment c','d.DiscussionID = c.DiscussionID')
+            ->where('c.InsertUserID', $userID)
+            ->groupBy('c.DiscussionID')
+            ->orderBy('d.DateLastComment', 'desc')
+            ->limit($limit, $offset)
+            ->get();
+
+        $sender->addDiscussionColumns($data);
+
+        return $data;
+    }
+
+    /**
+     * Gets number of discussions user has commented on.
+     *
+     * @param DiscussionModel $sender
+     * @param array $args
+     * @throws Exception if user is not logged in.
+     * @return int|false
+     */
+    public function discussionModel_getCountParticipated_create($sender) {
+        $userID = val(0, $sender->EventArguments);
+
+        if ($userID === null) {
+            if (!Gdn::session()->isValid()) {
+                throw new Exception(t('Could not get participated discussions for non logged-in user.'));
+            }
+            $userID = Gdn::session()->UserID;
+        }
+
+        $count = Gdn::sql()->select('c.DiscussionID','distinct','NumDiscussions')
+            ->from('Comment c')
+            ->where('c.InsertUserID', $userID)
+            ->groupBy('c.DiscussionID')
+            ->get();
+
+        $result = ($count instanceof Gdn_Dataset) ? $count->numRows() : false;
+        return $result;
+    }
+
+    /**
+     * Add navigation tab.
+     *
+     * @deprecated
+     * @param mixed $sender
+     */
+    public function addParticipatedTab($sender) {
+        $myParticipated = T('Participated Discussions');
+        $attributes = [];
+
+        if ($sender->RequestMethod == 'participated') {
+            $attributes['class'] = 'Active';
+        }
+
+        $result = wrap(
+            Anchor($myParticipated, '/discussions/participated', 'MyParticipated TabLink'),
+            'li',
+            $attributes
+        );
+        echo $result;
+    }
+
+    /**
+     * @param DiscussionsController $sender
+     */
+    public function discussionsController_afterDiscussionTabs_handler($sender) {
+        $this->addParticipatedTab($sender);
+    }
+
+    /**
+     * @param CategoriesController $sender
+     */
+    public function categoriesController_afterDiscussionTabs_handler($sender) {
+        $this->addParticipatedTab($sender);
+    }
+
+    /**
+     * @param DraftsController $sender
+     */
+    public function draftsController_afterDiscussionTabs_handler($sender) {
+        $this->addParticipatedTab($sender);
+    }
+
+    /**
+     * New navigation menu item.
+     *
+     * @since 2.1
+     * @param mixed $sender
+     */
+    public function base_afterDiscussionFilters_handler($sender) {
+        if (!Gdn::session()->checkPermission('Garden.SignIn.Allow')) {
+            return;
+        }
+
+        // Participated
+        $cssClass = 'Participated';
+        $controller = strtolower(Gdn::Controller()->ControllerName);
```

Comment:
Casening! You can use this: https://gist.github.com/DaazKu/fc20843e3f7bd2e0311194c0e70dada5